src/vcpkg/base/cofffilereader.cpp

```diff
@@ -239,113 +257,299 @@ namespace
+                    if (!read.has_value())
+                    {
+                        return std::move(read).error();
+                    }
+                }
+
+                // Object files shouldn't have optional headers, but the spec says we should skip over one if any
+                {
+                    auto seek = f.try_seek_to(coff_header.size_of_optional_header, SEEK_CUR);
+                    if (!seek.has_value())
+                    {
+                        return std::move(seek).error();
+                    }
+                }
+
+                // Read section headers
+                std::vector<SectionTableHeader> sections;
+                sections.resize(coff_signature.number_of_sections);
+                {
+                    auto read =
+                        f.try_read_all(sections.data(), sizeof(SectionTableHeader) * coff_signature.number_of_sections);
+                    if (!read.has_value())
+                    {
+                        return std::move(read).error();
+                    }
+                }
+
+                // Look for linker directive sections
+                for (auto&& section : sections)
+                {
+                    if (!(section.characteristics & SectionTableFlags::LinkInfo) ||
+                        memcmp(".drectve", &section.name, 8) != 0 || section.number_of_relocations != 0 ||
+                        section.number_of_line_numbers != 0)
+                    {
+                        continue;
+                    }
+
+                    // read the actual directive
+                    std::string directive_command_line;
+                    const auto section_offset = coff_base + section.pointer_to_raw_data;
+                    {
+                        auto seek = f.try_seek_to(section_offset);
+                        if (!seek.has_value())
+                        {
+                            return std::move(seek).error();
+                        }
+                    }
+
+                    directive_command_line.resize(section.size_of_raw_data);
+                    {
+                        auto read = f.try_read_all(directive_command_line.data(), section.size_of_raw_data);
+                        if (!read.has_value())
+                        {
+                            return std::move(read).error();
+                        }
+                    }
+
+                    if (Strings::starts_with(directive_command_line, StringLiteral{"\xEF\xBB\xBF"}))
+                    {
+                        // chop off the BOM
+                        directive_command_line.erase(0, 3);
+                    }
+
+                    for (auto&& directive : tokenize_command_line(directive_command_line))
+                    {
+                        auto insertion_point =
+                            std::lower_bound(directives.begin(), directives.end(), directive_command_line);
```

Comment:
The comment above is "`// used as sets because n is tiny`", so this was at least an intentional thing. I can extract add_set_vector though...

Reply:
https://github.com/BillyONeal/vcpkg-tool/commit/1d3ec7e3f445acb4cf436f56a969faa296cd8210 I tried implementing this but decided to switch to just use a set instead. Unlike MachineTypes where we know N is at most ~3, we don't know that for directives.

As part of that I also wrote https://github.com/microsoft/vcpkg-tool/pull/885 but didn't apply that here since I didn't do that.